pkg/ruler/rules/objectclient/rule_store.go

```diff
@@ -149,6 +149,29 @@ func (o *RuleStore) DeleteRuleGroup(ctx context.Context, userID string, namespac
 	return err
 }
 
+// DeleteNamespace deletes all the rule groups in the specified namespace
+func (o *RuleStore) DeleteNamespace(ctx context.Context, userID, namespace string) error {
+	ruleGroupObjects, _, err := o.client.List(ctx, generateRuleObjectKey(userID, namespace, ""))
```

Comment:
Implementation of `generateRuleObjectKey` relies on the fact that base64("") = "". For clarity, I would avoid using that assumption in `generateRuleObjectKey` method, because base64-decoding of strings like "=" or "==" also returns "", and it's not clear if encoding "" doesn't lead to one of those.

Reply:
Good shout, I found that a bit strange too but did not have a justification for a change. The case of `=` and `==` gives me that.